batch/batch/server/server.py

```diff
@@ -458,7 +459,8 @@ def __init__(self, batch_id, job_id, attributes, callback, userdata, user, alway
         incomplete_parent_ids = await db.jobs.get_incomplete_parents(*self.id)
         if self._state == 'Pending' and not incomplete_parent_ids:
             parents = [Job.from_record(record) for record in await db.jobs.get_parents(*self.id)]
-            if self.always_run or all(p.is_successful() for p in parents):
+            if (self.always_run or
+                (not self._cancelled and all(p.is_successful() for p in parents))):
```

Comment:
Should we also check that n_cancelled == 0?

Reply:
Should we check whether the job's batch has `n_cancelled > 0`? I'm not sure this is actually the right thing, because the CASCADE works one way, always incrementing, and so if one job fails, without always_run all others would fail (with cancelled) as well.

Still, if that could be made to work, say by decrementing n_cancelled, you could avoid doing the work in two places. 

In fact, without the decrement, I'm not sure n_cancelled can be relied on?